src/libs/wds-status.test.ts

```diff
@@ -0,0 +1,489 @@
+import { asMockedFn, renderHookInAct } from 'src/testing/test-utils';
+
+import { useWdsStatus } from './wds-status';
+
+type AjaxExports = typeof import('src/libs/ajax');
+jest.mock('src/libs/ajax', (): Partial<AjaxExports> => {
+  return { Ajax: jest.fn() };
+});
+
+type AjaxContract = ReturnType<typeof Ajax>;
+
+describe('useWdsStatus', () => {
+  const workspaceId = '6601fdbb-4b53-41da-87b2-81385f4a760e';
+
+  it('fetches Leo apps', async () => {
+    // Arrange
+    const listAppsV2 = jest.fn().mockResolvedValue([]);
+    const mockAjax: DeepPartial<AjaxContract> = {
+      Apps: {
+        listAppsV2,
+      },
+    };
+    asMockedFn(Ajax).mockReturnValue(mockAjax as AjaxContract);
+
+    // Act
+    await renderHookInAct(() => useWdsStatus({ workspaceId }));
+
+    // Assert
+    expect(listAppsV2).toHaveBeenCalledWith(workspaceId);
+  });
+
+  describe('if fetching Leo apps fails', () => {
+    it('returns unknown for all fields', async () => {
+      // Arrange
+      const mockAjax: DeepPartial<AjaxContract> = {
+        Apps: {
+          listAppsV2: jest.fn().mockRejectedValue(new Error('Something went wrong')),
+        },
+      };
+      asMockedFn(Ajax).mockReturnValue(mockAjax as AjaxContract);
+
+      // Act
+      const { result: hookReturnRef } = await renderHookInAct(() => useWdsStatus({ workspaceId }));
+
+      // Assert
+      expect(hookReturnRef.current.status).toEqual({
+        numApps: 'unknown',
+        appName: 'unknown',
+        appStatus: 'unknown',
+        proxyUrl: 'unknown',
+        wdsResponsive: 'unknown',
+        version: 'unknown',
+        wdsStatus: 'unresponsive',
+        wdsDbStatus: 'unknown',
+        wdsPingStatus: 'unknown',
+        wdsIamStatus: 'unknown',
+        defaultInstanceExists: 'unknown',
+      });
+    });
+  });
+
+  describe('if Leo apps are fetched successfully', () => {
+    describe('if no WDS app is present', () => {
+      it('returns number of apps and unknown for other fields', async () => {
+        // Arrange
+        const mockAjax: DeepPartial<AjaxContract> = {
+          Apps: {
+            listAppsV2: jest.fn().mockResolvedValue([]),
+          },
+        };
+        asMockedFn(Ajax).mockReturnValue(mockAjax as AjaxContract);
+
+        // Act
+        const { result: hookReturnRef } = await renderHookInAct(() => useWdsStatus({ workspaceId }));
+
+        // Assert
+        expect(hookReturnRef.current.status).toEqual({
+          numApps: '0',
+          appName: 'unknown',
+          appStatus: 'unknown',
+          proxyUrl: 'unknown',
+          wdsResponsive: 'unknown',
+          version: 'unknown',
+          wdsStatus: 'unresponsive',
+          wdsDbStatus: 'unknown',
+          wdsPingStatus: 'unknown',
+          wdsIamStatus: 'unknown',
+          defaultInstanceExists: 'unknown',
+        });
+      });
+    });
+  });
+
+  describe('if WDS app is present', () => {
+    const wdsApp: ListAppResponse = {
```

Comment:
This is what prompted the changes to app models. This is copied and pasted (with minor changes) from a dev API response.